CMakeLists.txt

```diff
@@ -36,7 +36,7 @@
 
 # Initial Setup
 
-CMAKE_MINIMUM_REQUIRED(VERSION 3.1.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.7)
```

Comment:
which feature of CMake we take from `>=3.7` ? Like hpc-conding-conventions repo, it would be great if we can make it work with `>= 3.5`.

Reply:
I told Omar CMake 3.7 is required in hpc-coding-conventions. But actually, 3.6 is sufficient only when `coreneuron_STATIC_ANALYSIS` CMake option is enabled.

Reply:
I do need 3.7 for the macro option parsing